webpack/scenes/RedHatRepositories/components/RepositorySetRepository.js

```diff
@@ -64,12 +66,27 @@ class RepositorySetRepository extends Component {
   }
 
   render() {
-    const { arch, releasever } = this.props;
+    const { arch, releasever, type } = this.props;
```

Comment:
Is it typical to define things like `yStreamHelp` which don't depend on the props inside of render?

Reply:
I'm not sure. XD @johnpmitsch do you know?

Reply:
We do this in other pages, but I'm not a huge fan of having so much logic in the `render()` function. I think these are evaluated every time `render()` is called and they should work fine to define this way outside of `render()`

Reply:
I will move it out. My reasoning for putting it in render is that it's the minimal scope of use for these functions.

Reply:
Where exactly should I define `yStreamHelp`? I tried adding it to the class as a method as well as defining it with `this.yStreamHelp =` in the constructor, but neither of those places are in scope of `render()`.

Reply:
At the top of this file, outside the class scope?

Reply:
its not obvious, but you have to do this:

```
diff --git a/webpack/scenes/RedHatRepositories/components/RepositorySetRepository.js b/webpack/scenes/RedHatRepositories/components/RepositorySetRepository.js
index fa7a6d8..9da8a1e 100644
--- a/webpack/scenes/RedHatRepositories/components/RepositorySetRepository.js
+++ b/webpack/scenes/RedHatRepositories/components/RepositorySetRepository.js
@@ -12,14 +12,6 @@ import api from '../../../services/api';
 
 const UNSPECIFIED_ARCH = 'Unspecified';
 
-const yStreamHelp = () => (
-  <span translate>
-    This repository is not suggested. Please see additional&nbsp;
-    <a href="https://access.redhat.com/articles/1586183">documentation</a>
-    &nbsp;prior to use.
-  </span>
-);
-
 class RepositorySetRepository extends Component {
   constructor(props) {
     super(props);
@@ -73,13 +65,21 @@ class RepositorySetRepository extends Component {
     };
   }
 
+  yStreamHelp = () => (
+    <span translate>
+      This repository is not suggested. Please see additional&nbsp;
+      <a href="https://access.redhat.com/articles/1586183">documentation</a>
+      &nbsp;prior to use.
+    </span>
+  );
+
   render() {
     const { arch, releasever, type } = this.props;
     const shouldDeemphasize = () => type !== 'kickstart' && yStream(releasever);
     const repositoryHeading = () => (
       <span>
         {arch} {releasever}
-        {shouldDeemphasize() ? (<FieldLevelHelp content={yStreamHelp()} />) : null}
+        {shouldDeemphasize() ? (<FieldLevelHelp content={this.yStreamHelp()} />) : null}
       </span>
     );
 
```

this allows you to define the method in the class and not do the `.bind(this)` dance in the constructor. Seems like we are moving to this pattern by the looks of recent React changes

Reply:
@johnpmitsch looks like @akofink got it working outside of the class :astonished: any preference on keeping it or going with your approach?

Reply:
would rather define it in the class, since that is the [pattern we are using elsewhere](https://git.io/fAZT8), the other functions in the render method can move as well. 

They do return JSX so an argument could be made to keep them inside `render()` (sorry I missed that they return JSX before), but I don't love crowding up render() with functions. Either way, definitely define these in the class.

Reply:
Don't worry, I'm going to move it where @johnpmitsch wants. Just distracted atm. :)